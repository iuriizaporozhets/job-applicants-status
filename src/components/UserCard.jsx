import React from 'react';
import styled from 'styled-components';

import { USER_STATUS, STATUS_CHANGE_ACTIONS, getNewUserStatus } from '../helpers';
import { capitalize } from '../utilities';

const UserCard = (props) => {
    const {user, className, changeUserStatus} = props;
    const {picture, status} = user;

    return (
        <div className={className}>
            <header>
                <UserPhotoWrapper>
                    <img src={picture.medium}/>
                </UserPhotoWrapper>
                <UserName>
                    {capitalize(user.name.first) + ' ' + capitalize(user.name.last)}
                </UserName>
            </header>
            <FooterStyled>
                { status !== USER_STATUS.APPLIED ?
                    <ButtonStyled
                        onClick={() => {
                            changeUserStatus(user, getNewUserStatus(status, STATUS_CHANGE_ACTIONS.BACK))
                        }}>&#8592;</ButtonStyled> :
                    null }
                { status !== USER_STATUS.HIRED ?
                    <ButtonStyled
                        onClick={() => {
                            changeUserStatus(user, getNewUserStatus(status, STATUS_CHANGE_ACTIONS.FORWARD))
                        }}>&#8594;</ButtonStyled> :
                    null
                }
            </FooterStyled>
        </div>
    );
};

const FooterStyled = styled.footer`
    display: flex;
    justify-content: space-between;
`;

const ButtonStyled = styled.button`height: 20px;
    width: 20px;
    border-radius: 50%;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    display: inline-flex;
    background-color: bisque;
    border: none;
`;

const UserPhotoWrapper = styled.div`
    border-radius: 50%;
    width: 70px;
    height: 70px;
    overflow: hidden;
`;

const UserName = styled.div`
    margin-left: 10px;
`;

const UserCardStyled = styled(UserCard)`
    width: 100%;
    height: 130px;
    padding: 10px;
    box-sizing: border-box;
    border: 2px solid bisque;
    margin-top: 8px;
    
    header {
        display: flex;
        align-items: center;
    }
    
    footer {
        margin-top: 10px;
    }
`;

export default UserCardStyled;
