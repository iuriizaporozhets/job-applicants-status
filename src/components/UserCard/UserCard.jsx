import React from 'react';
import styled from 'styled-components';

import { USER_STATUS, STATUS_CHANGE_ACTIONS } from '../../constants';
import { getNewUserStatus } from '../../helpers';

const UserCard = (props) => {
	const {user, className, changeUserStatus} = props;
	const {image, status} = user;

	return (
		<div className={className}>
			<header>
				<UserPhotoWrapper>
					<img src={image}/>
				</UserPhotoWrapper>
				<UserInfo>
					<div>{user.name}</div>
					<div className="user-city">{user.city}</div>
				</UserInfo>
			</header>
			<FooterStyled>
				{ status !== USER_STATUS.APPLIED ?
					<ButtonStyled
						className="back-button"
						onClick={() => {
							changeUserStatus(user, getNewUserStatus(status, STATUS_CHANGE_ACTIONS.BACK));
						}}>&#8592;</ButtonStyled> :
					null }
				{ status !== USER_STATUS.HIRED ?
					<ButtonStyled
						className="forward-button"
						onClick={() => {
							changeUserStatus(user, getNewUserStatus(status, STATUS_CHANGE_ACTIONS.FORWARD));
						}}>&#8594;</ButtonStyled> :
					null
				}
			</FooterStyled>
		</div>
	);
};

const UserInfo = styled.div`
	margin-left: 10px;
	
	.user-city {
		font-size: 14px;
		color: #7d7d7d;
	}
`;

const FooterStyled = styled.footer`
    display: flex;
    justify-content: space-between;
`;

const ButtonStyled = styled.button`height: 20px;
    width: 20px;
    border-radius: 50%;
    cursor: pointer;
    align-items: center;
    justify-content: center;
    display: inline-flex;
    background-color: bisque;
    border: none;
    
    &:only-child {
    	&.back-button {
    		margin-right: auto;
    	}
    	
    	&.forward-button {
    		margin-left: auto;
    	}
    }
`;

const UserPhotoWrapper = styled.div`
    border-radius: 50%;
    width: 70px;
    height: 70px;
    overflow: hidden;
`;

const UserCardStyled = styled(UserCard)`
    width: 100%;
    height: 130px;
    padding: 10px;
    box-sizing: border-box;
    border: 2px solid bisque;
    margin-top: 8px;
    
    header {
        display: flex;
        align-items: center;
    }
    
    footer {
        margin-top: 10px;
    }
`;

export default UserCardStyled;
