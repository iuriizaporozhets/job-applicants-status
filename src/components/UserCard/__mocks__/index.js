const user = {
	city: 'Londonderry',
	id: 'ZK 54 86 60 O',
	image: 'https://randomuser.me/api/portraits/med/women/96.jpg',
	name: 'Martha Taylor',
	status: 'APPLIED',
};

export { user };