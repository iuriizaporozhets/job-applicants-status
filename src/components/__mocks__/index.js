const user = {
    gender: 'female',
    name: {
        first: 'abbie',
        last: 'mckinney',
    },
    location: {
        city: 'london',
    },
    picture: {
        medium: 'https://randomuser.me/api/portraits/med/women/18.jpg',
    },
    status: 'APPLIED'
};

const filter = {
    type: 'NAME',
    filterQuery: 'someQuery'
};

export { user, filter };