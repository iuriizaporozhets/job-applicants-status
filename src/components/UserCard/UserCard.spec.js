import React from 'react';
import UserCard from './UserCard';
import renderer from 'react-test-renderer';
import { user } from './__mocks__';

describe('UserCard > ', () => {
	let wrapper;
	const userCardProps = {
		user,
		className: 'className',
		changeUserStatus: jest.fn(),
	};

	beforeEach(() => {
		wrapper = renderer.create(<UserCard {...userCardProps} />);
	});

	afterEach(() => {
		wrapper.unmount();
	});

	it('should match snapshot', () => {
		const tree = wrapper.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('APPLIED user should contain 1 button', () => {
		expect(wrapper.root.findAllByType('button')).toHaveLength(1);
	});
});
