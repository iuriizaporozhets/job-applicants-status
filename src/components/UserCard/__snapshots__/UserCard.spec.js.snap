// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard >  should match snapshot 1`] = `
<div
  className="className sc-ifAKCX ebFgfC"
>
  <header>
    <div
      className="sc-bxivhb jYKomR"
    >
      <img
        src="https://randomuser.me/api/portraits/med/women/96.jpg"
      />
    </div>
    <div
      className="sc-bdVaJa hNlrIX"
    >
      <div>
        Martha Taylor
      </div>
      <div
        className="user-city"
      >
        Londonderry
      </div>
    </div>
  </header>
  <footer
    className="sc-bwzfXH liwCWi"
  >
    <button
      className="forward-button sc-htpNat hvivgI"
      onClick={[Function]}
    >
      â†’
    </button>
  </footer>
</div>
`;
